shared/engine/index.js.flow

```diff
@@ -0,0 +1,37 @@
+// @flow
+// Sep. engine file so we can avoid import cycles w/ flow-types.js
```

Comment:
flow-types pulls in engine which pulls in flow-types etc, lets not have that happen